packages/cli-config-apple/src/tools/runCodegen.ts

```diff
@@ -0,0 +1,35 @@
+import fs from 'fs';
+import path from 'path';
+import execa from 'execa';
+
+interface CodegenOptions {
+  root: string;
+  platform: string;
+}
+
+async function runCodegen(options: CodegenOptions): Promise<void> {
+  if (fs.existsSync('build')) {
+    fs.rmSync('build', {recursive: true});
+  }
+
+  const reactNativePath = path.dirname(
+    require.resolve('react-native', {paths: [process.cwd()]}),
+  );
```

Comment:
we could use `reactNativePath` from config

Reply:
https://github.com/react-native-community/cli/pull/2600/commits/48c4d92593376cdcbf589ac89b652460eea45bca